scripts/scr_start_load/scr_start_load.gml

```diff
@@ -1,547 +1,122 @@
-    obj_ini.num_of_cruisers = num_of_cruisers;
-
-    obj_ini.max_bl = max_bl;
-    obj_ini.max_fr = max_fr;
-
-    obj_ini.barge = barge;
-    obj_ini.frigate = frigate;
-
-    obj_ini.remove_size = remove_size;
-
-    // Final check to ensure no companies are left unsupported
-    repeat (10) {
-        if (obj_ini.man_size > 0 && obj_ini.ship_carrying[ship] == 0) {
-            obj_ini.man_size = 0;
-        }
-        ship += 1;
-    }
-  }
- }
-}
+		}
+					//if company won't fit onto ship
+		if ((obj_ini.ship_carrying[ship] - _company_size) > obj_ini.ship_capacity[ship]){
+			ship_fit = false;
+		}		
+		//if entire company won't fit on ship test to see if there is any ship in the fleet the company will fit on;
+		 if (ship_fit == false){
+			 for (var ship_loop =  ship+1; ship_loop< array_length(obj_ini.ship_carrying);ship_loop++;){
+				  if ((obj_ini.ship_carrying[ship_loop] - _company_size) <= obj_ini.ship_capacity[ship_loop]){
+					  //load marines
+					  for (var m = 0; m <array_length(company_loader);m++;){
+						   load_marine(company_loader[m][0], company_loader[m][1], ship_loop, company_loader[m][2]);
+					  }
+					  //load vehicles
+					  for (var m = 0; m <array_length(company_vehicle);m++;){
+						   load_vehicles(company_vehicle[m][0], company_vehicle[m][1], ship_loop,  company_vehicle[m][2]);
+					  }
+					  ship_fit = true;
+					  break;
+
+				  }
+			 }
+			 // if there are no ships that will hold the entire company loop all ships and jam pac the fuckers in in
+			 if (ship_fit == false){
+				 for (var ship_loop = ship+1; ship_loop<array_length(obj_ini.ship_carrying); ship_loop++;){
+				 if (obj_ini.ship_carrying[ship_loop] < obj_ini.ship_capacity[ship_loop]){
+					 ship_has_space = true;	
+					 // new arrays that will not contain troops that get loaded
+					 var comp_edit =[];
+					 var veh_edit = [];
+					  for (var m = 0; m <array_length(company_loader);m++;){
+						  if ((obj_ini.ship_carrying[ship_loop] - company_loader[m][2]) <= obj_ini.ship_capacity[ship_loop]){
+							load_marine(company_loader[m][0], company_loader[m][1], ship_loop, company_loader[m][2]);
+						  } else {array_push(comp_edit, company_loader[m])}
+
+						  if (obj_ini.ship_carrying[ship_loop] = obj_ini.ship_capacity[ship_loop]){
+							   ship_has_space = false;
+							   break;
+						  }
+					  }
+					  for (var m = 0; m <array_length(company_vehicle);m++;){
+						  if ((obj_ini.ship_carrying[ship_loop] - company_vehicle[m][2]) <= obj_ini.ship_capacity[ship_loop]){
+							load_vehicles(company_vehicle[m][0], company_vehicle[m][1], ship_loop, company_vehicle[m][2]);
+						  } else {array_push( veh_edit, company_vehicle[m])}
+						  if (obj_ini.ship_carrying[ship_loop] = obj_ini.ship_capacity[ship_loop]){
+							   ship_has_space = false;
+							   break;
+						  }
+					  }					  
+					   company_loader= comp_edit;
+					   company_vehicle = veh_edit;
+				 }
+				 }
+			 }
+		 } else{

```

Comment:
this is case when a single ship can fit all guys?

Reply:
yeah that's the one could be reordered to go before other stuff if more readable?

Reply:
usually we order if-elses by probability of happening, unless the if-else is simple, then go for readability. This one is just a simple bool check, so it can stay, unless you prefer to order. But it doesn't really matter in this case.